### ISSUE TITLE
Global Retention Policies

### ISSUE THREAD

**Issue by adambkaplan** [tektoncd/results#459]
## Feature Request

Define a global data retention policy feature for a Tekton Results. When enabled, a CronJob will regularly search for Results, Records, and Logs in storage and issue API calls to delete the data older than a specified duration.

## Use Cases

- Prune data that is obsolete.
- See discussion in https://github.com/tektoncd/results/issues/221#issuecomment-1538685843_

**Comment by khrm** [comment]
This and the pruner function provided by the operators are similar. @vdemeester 

Is the operator the suitable place for this or the results? If it's latter, then we should provide a way to run this pruner without the storage capability of results.

**Comment by sayan-biswas** [comment]
@khrm Results already have pruning mechanism in the controller, this I guess is more targeted in deleting the data from the database and/or S3.

**Comment by adambkaplan** [comment]
> this I guess is more targeted in deleting the data from the database and/or S3.

That is correct. And for this particular issue, the scope should be narrowed to a global, default retention for all Results records.

**Comment by adambkaplan** [comment]
> Is the operator the suitable place for this or the results? If it's latter, then we should provide a way to run this pruner without the storage capability of results.

My inclination is Results - perhaps as a component in the `config` directory. @khrm - can you please clarify what you mean by "without the storage capability"?

**Comment by adambkaplan** [comment]
/lifecycle frozen

This is something that should be considered on our long-term roadmap.

**Comment by enarha** [comment]
@khrm please confirm we can close this one
